RecoVertex/BeamSpotProducer/plugins/BeamSpotCompatibilityChecker.cc

```diff
@@ -0,0 +1,242 @@
+// -*- C++ -*-
+//
+// Package:    RecoVertex/BeamSpotProducer
+// Class:      BeamSpotCompatibilityChecker
+//
+/**\class BeamSpotCompatibilityChecker BeamSpotCompatibilityChecker.cc RecoVertex/BeamSpotProducer/plugins/BeamSpotCompatibilityChecker.cc
+
+ Description: [one line class summary]
+
+ Implementation:
+     [Notes on implementation]
+*/
+//
+// Original Author:  Marco Musich
+//         Created:  Thu, 23 Apr 2020 09:00:45 GMT
+//
+//
+
+// system include files
+#include <memory>
+
+// user include files
+#include "CondFormats/BeamSpotObjects/interface/BeamSpotObjects.h"
+#include "CondFormats/DataRecord/interface/BeamSpotObjectsRcd.h"
+#include "DataFormats/BeamSpot/interface/BeamSpot.h"
+#include "DataFormats/Common/interface/Handle.h"
+#include "DataFormats/Scalers/interface/BeamSpotOnline.h"
+#include "FWCore/Framework/interface/ESHandle.h"
+#include "FWCore/Framework/interface/ESWatcher.h"
+#include "FWCore/Framework/interface/Event.h"
+#include "FWCore/Framework/interface/EventSetup.h"
+#include "FWCore/Framework/interface/Frameworkfwd.h"
+#include "FWCore/Framework/interface/MakerMacros.h"
+#include "FWCore/Framework/interface/global/EDAnalyzer.h"
+#include "FWCore/MessageLogger/interface/MessageLogger.h"
+#include "FWCore/ParameterSet/interface/ConfigurationDescriptions.h"
+#include "FWCore/ParameterSet/interface/ParameterSet.h"
+#include "FWCore/ParameterSet/interface/ParameterSetDescription.h"
+#include "FWCore/Utilities/interface/ESGetToken.h"
+#include "FWCore/Utilities/interface/InputTag.h"
+
+//
+// class declaration
+//
+
+// ancillary struct to check compatibility
+namespace bscheck {
+
+  struct Significance {
+    Significance(const double& A, const double& B, const double& ErrA, const double& ErrB)
+        : m_A(A), m_B(B), m_ErrA(ErrA), m_ErrB(ErrB) {
+      if (m_ErrA == 0 && m_ErrB == 0) {
+        edm::LogError("LogicalError") << "Can't calculate significance when both errors are zero!" << std::endl;
+      }
+    }
+
+    float getSig(bool verbose) {
+      if (verbose) {
+        edm::LogInfo("BeamSpotCompatibilityChecker")
+            << "A= " << m_A << "+/-" << m_ErrA << "    "
+            << "B= " << m_B << "+/-" << m_ErrB << " | delta=" << std::abs(m_A - m_B) << "+/-"
+            << std::sqrt((m_ErrA * m_ErrA) + (m_ErrB * m_ErrB)) << std::endl;
+      }
+      return std::abs(m_A - m_B) / std::sqrt((m_ErrA * m_ErrA) + (m_ErrB * m_ErrB));
```

Comment:
since only `sqrt(eA^2+eB^2)` is used, better precompute it in the constructor

Reply:
fixed in the last push.